src/software/new_geom/linear_spline2d.cpp

```diff
@@ -0,0 +1,103 @@
+#include "software/new_geom/linear_spline2d.h"
+
+#include <algorithm>
+#include <sstream>
+
+LinearSpline2d::LinearSpline2d(const std::vector<Point>& points) : knots(points)
+{
+    initLinearSegments(points);
+}
+
+LinearSpline2d::LinearSpline2d(const std::initializer_list<Point>& points) : knots(points)
+{
+    initLinearSegments(points);
```

Comment:
nit: because the initializer list can be implicitly cast to a vector, why not put the implementation of `initLinearSegments` in the constructor that takes the vector, and then just call that constructor from here. 

Then don't have to remember to update initialization lists in two places / constructors if things change, and don't need the extra function.

Reply:
Yeah, so we can't do this because passing the initializer list to the constructor at that point is ambiguous, and the compiler will complain about a cycle.